app/api/routes.py

```diff
@@ -332,7 +338,7 @@ def update_resource(id, json, db):
         return redirect('/404')
 
     langs, categ = get_attributes(json)
-    index_object = {'objectID': resource.id}
+    index_object = {'objectID': id}
```

Comment:
@aaron-suarez this and the partial _update_object change lets it update properly. The change to category also lets it save the right category